src/FormHandler/FormSubmitProcessor.php

```diff
@@ -27,18 +27,26 @@
     private $on_failure;
 
     /**
+     * @var callable
+     */
+    private $form_submit_processor;
+
+    /**
      * @param FormInterface $form
      * @param callable|null $on_success
      * @param callable|null $on_failure
+     * @param callable|null $form_submit_processor
      */
     public function __construct(
         FormInterface $form,
         callable $on_success = null,
-        callable $on_failure = null
+        callable $on_failure = null,
+        callable $form_submit_processor = null
     ) {
-        $this->on_success = $on_success;
-        $this->on_failure = $on_failure;
-        $this->form       = $form;
+        $this->on_success            = $on_success;
+        $this->on_failure            = $on_failure;
+        $this->form                  = $form;
+        $this->form_submit_processor = $form_submit_processor;
```

Comment:
This was already in the wrong order, but can you move the form assignment to the top? Makes sense to have them in the same order as the constructor arguments